### ISSUE TITLE
Replace ebib-db-list-keys with ebib--list-keys in certain places

### ISSUE THREAD

**Issue by myshevchuk** [joostkremers/ebib#174]
Hi! I've just started to use Ebib and find it simply amazing. Thank you very much for your hard work!

There's a couple of issues I encountered.

1. When saving filtered entries into a new file, the whole database is saved rather than the filtered entries.

`ebib-utils.el` on line 1804
```elisp
( let ((sorted-list (sort (ebib-db-list-keys ebib--cur-db) #'string<)))
...
```
should read 

```elisp
( let ((sorted-list (sort (ebib--list-keys ebib--cur-db) #'string<)))
...
```
2. The same happens when marking all filtered entries - the whole database is marked.

`ebib-db.el` on line 511
``` elisp
...
    (setf (ebib-db-val 'marked-entries db) (ebib-db-list-keys db)))))
```
Should read
``` elisp
...
    (setf (ebib-db-val 'marked-entries db) (ebib--list-keys db)))))
```

Cheers,
Michael

**Comment by joostkremers** [comment]
Thanks for pointing out these bugs. :grimacing: 

They're not as easy to fix as you suggest, though, because that would introduce circular dependencies (`ebib.el` depends on both `ebib-utils.el` and `ebib-db.el`). Moving `ebib--list-keys` to `ebib-db.el` would solve the problem, but that's not the appropriate file for that function. And since `ebib-utils.el` depends on `ebib-db.el`, moving `ebib--list-keys` to `ebib-utils.el` also won't work...

I have no time to devote to this right now, hopefully in a day or two.

**Comment by myshevchuk** [comment]
Thank you for a quick reply. I've applied the suggested patch in my local installation and am now happily using Ebib filtering 😀. No loading problems so far. I guess it only matters for the byte compiler, although mine doesn't complain.

Please take your time.

**Comment by joostkremers** [comment]
Yeah, `ebib--list-keys` is a function, so it shouldn't lead to many issues. The byte compiler will complain if you compile Ebib when it's not already loaded, but normally that'll be it. But you never know when something like this is gonna byte you in the behind, so I prefer to avoid circular dependencies if I can.

**Comment by joostkremers** [comment]
This issue should now be fixed. I moved `ebib--format-database-as-bibtex` to `ebib.el`, since it's only used there anyway, which makes it possible to call `ebib--list-keys` instead of `ebib-db-list-keys`.

The call in `ebib-db.el` was handled differently: instead of passing a symbol to mark all entries, I now pass a list of entries to be marked, so that the caller can create the list.

The same issue also existed with `ebib-db-unmark-entry`, BTW, which I fixed in the same way.

Thanks again for reporting these bugs. One wonders how they went unnoticed for so long... :worried:
